src/platform_linux/platform.c

```diff
@@ -103,16 +111,21 @@ platform_buffer_getaddr(const buffer_handle *bh)
    return bh->addr;
 }
 
+/*
+ * platform_buffer_destroy() - Destroy the large buffer created using mmap()
+ * Free all associated memory. buffer_handle *bh_in is NULL'ed out on exit.
+ */
 platform_status
-platform_buffer_destroy(buffer_handle *bh)
+platform_buffer_destroy(platform_heap_id heap_id, buffer_handle **bh_in)
 {
-   int ret;
+   buffer_handle *bh = *bh_in;
+   int            ret;
    ret = munmap(bh->addr, bh->length);
    if (ret) {
       return CONST_STATUS(errno);
    }
 
-   platform_free(platform_get_heap_id(), bh);
+   platform_free(heap_id, *bh_in);
```

Comment:
Interface changed (for the same reasons as for the `create()` method) so that we can use the right `heap_id` while freeing memory. Otherwise, default `platform_get_heap_id()` may be NULL, and we may end up not freeing the memory with the proper heap-ID.